producer/src/test/kotlin/streams/integrations/KafkaEventRouterCompactionStrategyTSE.kt

```diff
@@ -612,4 +612,122 @@ class KafkaEventRouterLogCompactionTSE : KafkaEventRouterBaseTSE() {
+    }
+
+    // we create a topic with kafka.streams.log.compaction.strategy=compact
+    // after that, we create and update some nodes and relationships
+    // finally, we check that each node/relationship has no records spread across multiple partitions but only in a single partition
+    private fun createTopicAndEntitiesAndAssertPartition(withConstraints: Boolean,
+                                                         firstExpectedKey: Any,
+                                                         secondExpectedKey: Any,
+                                                         thirdExpectedKey: Any,
+                                                         fourthExpectedKey: Any) {
+        val relType = "KNOWS"
+
+        // we create a topic with strategy compact
+        val topic = UUID.randomUUID().toString()
+        val sourceTopics = mapOf("streams.source.topic.nodes.$topic" to "Person{*}",
+                "streams.source.topic.relationships.$topic" to "$relType{*}",
+                "kafka.num.partitions" to "10" )
+        // we optionally create a constraint for Person.name property
+        val queries = if(withConstraints) listOf("CREATE CONSTRAINT ON (p:Person) ASSERT p.name IS UNIQUE") else null
+
+        initDbWithLogStrategy(TopicConfig.CLEANUP_POLICY_COMPACT, sourceTopics, queries)
+        KafkaLogCompactionTestCommon.createTopic(topic, bootstrapServerMap, 10, false)
+
+        kafkaConsumer.subscribe(listOf(topic))
+
+        // we create some nodes and rels
+        db.execute("CREATE (:Person {name:'Pippo', surname: 'Pippo_1'})")
+        db.execute("CREATE (:Person:Other {name:'Pluto', surname: 'Pluto_1'})")
+        db.execute("CREATE (:Person:Other {name:'Paperino', surname: 'Paperino_1'})")
+        db.execute("CREATE (:Person {name:'Foo'})")
+        db.execute("CREATE (:Person:Other {name:'Bar'})")
+        db.execute("CREATE (:Person {name:'Baz'})")
+        db.execute("CREATE (:Person {name:'Baa'})")
+        db.execute("CREATE (:Person {name:'One'})")
+        db.execute("CREATE (:Person {name:'Two'})")
+        db.execute("CREATE (:Person {name:'Three'})")
+        db.execute("""
+                |MATCH (pippo:Person {name:'Pippo'})
+                |MATCH (pluto:Person {name:'Pluto'})
+                |MERGE (pippo)-[:KNOWS]->(pluto)
+            """.trimMargin())
+        db.execute("""
+                |MATCH (foo:Person {name:'Foo'})
+                |MATCH (bar:Person {name:'Bar'})
+                |MERGE (foo)-[:KNOWS]->(bar)
+            """.trimMargin())
+        db.execute("""
+                |MATCH (one:Person {name:'One'})
+                |MATCH (two:Person {name:'Two'})
+                |MERGE (one)-[:KNOWS]->(two)
+            """.trimMargin())
+
+        // we update 4 nodes and 2 rels
+        db.execute("MATCH (p:Person {name:'Pippo'}) SET p.surname = 'Pippo_update'")
+        db.execute("MATCH (p:Person {name:'Pippo'}) SET p.address = 'Rome'")
```

Comment:
```suggestion
        db.execute("""
          |MATCH (p:Person {name:'Pippo'})
          |SET p.surname = 'Pippo_update',
          |p.address = 'Rome'
        """.trimMargin())
```

Reply:
anche in quelli sotto

Reply:
aspetta se c'è un motivo preciso per cui fai le update separate allora spiegami e aggiungi semplicemente un commento al codice...

Reply:
Semplicemente l'ho fatto più volte per "fare numero", cioè per avere più record così da verificare in seguito che, a parità di entità, tutti i record appartengono alla stessa partizione. Con pochi record è più randomica la cosa, perché magari potrebbero andare a finire nella stessa partizione per puro caso